src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -186,46 +186,57 @@ SendCommandListToWorkerInSingleTransaction(char *nodeName, int32 nodePort, char
 							   nodeName, nodePort, nodeUser)));
 	}
 
-	/* start the transaction on the worker node */
-	queryResult = PQexec(workerConnection, "BEGIN");
-	if (PQresultStatus(queryResult) != PGRES_COMMAND_OK)
+	PG_TRY();
 	{
-		ReraiseRemoteError(workerConnection, queryResult);
-	}
+		/* start the transaction on the worker node */
+		queryResult = PQexec(workerConnection, "BEGIN");
+		if (PQresultStatus(queryResult) != PGRES_COMMAND_OK)
+		{
+			ReraiseRemoteError(workerConnection, queryResult);
+		}
 
-	PQclear(queryResult);
+		PQclear(queryResult);
 
-	/* iterate over the commands and execute them in the same connection */
-	foreach(commandCell, commandList)
-	{
-		char *commandString = lfirst(commandCell);
-		ExecStatusType resultStatus = PGRES_EMPTY_QUERY;
+		/* iterate over the commands and execute them in the same connection */
+		foreach(commandCell, commandList)
+		{
+			char *commandString = lfirst(commandCell);
+			ExecStatusType resultStatus = PGRES_EMPTY_QUERY;
+
+			queryResult = PQexec(workerConnection, commandString);
```

Comment:
We could maybe add a CHECK_FOR_INTERRUPTS before the PQexec to at least have some earlier cancellation opportunities.